feat: introduce PollManager

Antreesy opened the pull request:
### ☑️ Resolves

* Fix #13439 
* Inroduce wrapper for PollEditor and PollDraftsHandler, allow to open them from any place:
  * chat attachments
  * shared items
  * conversation settings

## 🖌️ UI Checklist

### 🖼️ Screenshots / Screencasts

Drafts | Editor
-- | --
<img width="523" alt="image" src="https://github.com/user-attachments/assets/82b9fb31-cc04-4095-a807-ba8c5ec55ee2" /> | <img width="605" alt="image" src="https://github.com/user-attachments/assets/e1534a25-147b-4a32-bbca-b6ea7e20c1ee" />

### 🚧 Tasks

- [x] It's possible to be in conversation1, open settings for conversation2 and create draft/poll in conversation2. Considered OK or we should block it?

### 🏁 Checklist

- [x] 🌏 Tested with different browsers / clients:
  - [x] Chromium (Chrome / Edge / Opera / Brave)
  - [ ] Firefox
  - [ ] Safari
  - [ ] Talk Desktop
  - [ ] Not risky to browser differences / client
- [ ] ⛑️ Tests are included or not possible
- [ ] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

DorraJaouad commented on src/components/ConversationSettings/LobbySettings.vue:
```diff
@@ -182,6 +197,11 @@ export default {
 		},
+
+		canCreatePollDrafts() {
+			return hasTalkFeature(this.token, 'talk-polls-drafts') && this.$store.getters.isModerator
```
`this.$store.getters.isModerator` is checking moderation against the token from root 💀

DorraJaouad commented on src/components/PollViewer/PollEditor.vue:
```diff
@@ -223,14 +225,15 @@ async function createPoll() {
  * Pre-fills form from the draft
  * @param id poll draft ID
- * @param isAlreadyOpened poll draft ID
+ * @param fromDrafts whether editor was opened from drafts handler
```
`fromDrafts` can  be used to hide "Browse poll drafts" button as it feels repetitive and user would intuitively pick from the poll drafts handler rather then browsing from that button (when it is opened in lobby settings or shared Items tab).

DorraJaouad commented on src/components/PollViewer/PollEditor.vue:
Also, not related but todo, hide/disable export poll button when the poll inputs are all empty

DorraJaouad commented on src/components/PollViewer/PollDraftHandler.vue:
Minor: we need a loading state while the poll drafts are being fetched.

DorraJaouad commented on src/components/PollViewer/PollEditor.vue:
```diff
@@ -289,7 +292,7 @@ async function createPollDraft() {
  */
 function openPollDraftHandler() {
-	EventBus.emit('poll-drafts-open')
+	EventBus.emit('poll-drafts-open', {})
```
Selector needs to be passed here, as the editor can be opened in settings (create a new poll) so clicking "back" mounts the drafts handler correctly

DorraJaouad commented on src/components/PollViewer/PollManager.vue:
```diff
@@ -0,0 +1,71 @@
+import { hasTalkFeature } from '../../services/CapabilitiesManager.ts'
+import { EventBus } from '../../services/EventBus.ts'
+
+const store = useStore()
```
not used

DorraJaouad commented on src/components/PollViewer/PollManager.vue:
```diff
@@ -0,0 +1,71 @@
+		showPollDraftHandler.value = false
+	})
+}
+
```
```suggestion
```

DorraJaouad commented:
> It's possible to be in conversation1, open settings for conversation2 and create draft/poll in conversation2. Considered OK or we should block it?

I'd say to not to block it, as the other conversation settings are also accessible, so we keep the same pattern

Antreesy commented:
> I'd say to not to block it, as the other conversation settings are also accessible, so we keep the same pattern

Adjusted button text to indicate, where are you creating poll
<img width="605" alt="image" src="https://github.com/user-attachments/assets/e1534a25-147b-4a32-bbca-b6ea7e20c1ee" />

DorraJaouad commented on src/views/MainView.vue:
```diff
@@ -10,6 +10,7 @@
 			<TopBar :is-in-call="isInCall" />
 			<CallView v-if="isInCall" :token="token" />
 			<ChatView v-else />
+			<PollManager />
```
It should also be accessible when route doesn't include a token (like duplicate session view etc) as long as the conversation settings are accessible there

DorraJaouad commented on src/components/PollViewer/PollManager.vue:
```diff
@@ -0,0 +1,77 @@
+	const { participantType, type } = store.getters.conversation(token.value) ?? {}
+	// TODO: getters.isModerator should accept token
+	return hasTalkFeature(token.value, 'talk-polls-drafts')
+		&& (participantType === PARTICIPANT.TYPE.OWNER || participantType === PARTICIPANT.TYPE.MODERATOR)
+		&& (type === CONVERSATION.TYPE.GROUP || type === CONVERSATION.TYPE.PUBLIC)
```
`GUEST_MODERATOR` no?

DorraJaouad commented on src/components/PollViewer/PollEditor.vue:
```diff
@@ -165,6 +168,12 @@ const pollForm = reactive<createPollParams>({
 const isFilled = computed(() => Boolean(pollForm.question) && pollForm.options.filter(option => Boolean(option)).length >= 2)
+const createPollLabel = computed(() => {
+	return isOpenedFromDraft.value
+		? t('spreed', 'Create poll in {name}', { name: store.getters.conversation(props.token).displayName },
```
this will also be shown in drafts from shared Items

Antreesy replied:
repalced with `route.params.token !== props.token`, to match if a currently open conversation

DorraJaouad commented on src/enums.ts:
```diff
@@ -0,0 +1,23 @@
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+enum POLL_STATUS {
+	OPEN = 0,
+	CLOSED = 1,
+	DRAFT = 2,
+}
```
Interesting

Antreesy commented:
Rebased:
- moved <PollManager /> from MainView.vue to App.vue
- align moderator permissions check with backend
- useRoute for comparing current token and createPoll destination
- fix 'updateConversationSettingsToken' when closing conv.settings
- dropped enums

Antreesy commented on src/enums.ts:
then we could do it later with constants.js -> constants.ts

DorraJaouad commented on src/components/PollViewer/PollEditor.vue:
```diff
@@ -165,6 +170,12 @@ const pollForm = reactive<createPollParams>({
 
 const isFilled = computed(() => Boolean(pollForm.question) && pollForm.options.filter(option => Boolean(option)).length >= 2)
+const createPollLabel = computed(() => {
+	return route.params.token !== props.token
```
You can use `store.getters.getToken` as the store is imported anyway.

DorraJaouad reviewed (approved):
Tested, all good 🦅

Antreesy commented:
> aa wait it doesn't work in Files Talk chat tab

Fixed with the last review change

DorraJaouad reviewed (approved):
Everything works
